secp256r1/src/test.rs

```diff
@@ -0,0 +1,38 @@
+#![cfg(test)]
+
+use super::*;
+use soroban_sdk::{bytesn, Address, Env};
+
+const WASM: &[u8] =
+    include_bytes!("../target/wasm32-unknown-unknown/release/soroban_secp256r1_contract.wasm");
+
+#[test]
+fn test_native() {
+    let env = Env::default();
+    let contract_id = env.register_contract(None, Contract);
+    test(&env, &contract_id);
+}
+
+#[test]
+fn test_wasm() {
+    let env = Env::default();
+    let contract_id = env.register_contract_wasm(None, WASM);
+    test(&env, &contract_id);
+}
+
+fn test(env: &Env, contract_id: &Address) {
+    // Set an unlimited budget because it'll fail with the default budget.
+    env.budget().reset_unlimited();
```

Comment:
When running the `test_wasm` it fails if the budget is the default budget, even if I reset the budget just before calling verify. I only seem to be able to get it to pass if I first reset to an unlimited budget. cc @graydon @tomerweller